packages/ember-metal/lib/streams/key-stream.js

```diff
@@ -25,6 +25,12 @@ let KeyStream = BasicStream.extend({
 
   compute() {
     var object = this.sourceDep.getValue();
+    const type = typeof object;
+
+    if (type === 'boolean' || type === 'string') {
```

Comment:
I suspect this expression should also live as an assertion within `get`, it may help catch such issues